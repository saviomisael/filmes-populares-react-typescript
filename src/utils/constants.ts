export const IMAGE_URL = process.env.REACT_APP_IMAGE_URL;
