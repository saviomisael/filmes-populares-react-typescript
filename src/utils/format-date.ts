export const formatDate = (date: Date) =>
  date.toLocaleString('pt-BR', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });
