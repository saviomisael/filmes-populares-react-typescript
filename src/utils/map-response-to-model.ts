import { TvSerie } from 'models/TvSerie';
import { TvSerieResponse } from 'services/responses/TvSerieResponse';

const mapSingleResponseToTvSerieModel = (response: TvSerieResponse) =>
  new TvSerie(response.id, response.name, response.vote_average, response.poster_path);

export const mapResponseToTvSerieModel = (response: TvSerieResponse[]) =>
  response.map((x) => mapSingleResponseToTvSerieModel(x));
