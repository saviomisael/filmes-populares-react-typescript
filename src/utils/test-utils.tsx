import { render } from '@testing-library/react';
import { MemoryRouter, Switch, Route } from 'react-router-dom';
import { BaseStyles } from 'styles/base';
import { ThemeProvider } from 'styled-components';
import { theme } from 'styles/theme';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { storeMock } from 'mocks/configure-store-mock';

type optionsWithRouter = {
  initialRoutes?: string[];
  pathname: string;
};

export const renderTheme = (children: ReactNode) =>
  render(
    <ThemeProvider theme={theme}>
      <BaseStyles />
      {children}
    </ThemeProvider>,
  );

export const renderWithStoreProvider = (children: ReactNode) =>
  render(
    <Provider store={storeMock}>
      <ThemeProvider theme={theme}>
        <BaseStyles />
        {children}
      </ThemeProvider>
    </Provider>,
  );

export const renderWithRouter = (children: ReactNode, options: optionsWithRouter) => {
  const initialRoutes = options.initialRoutes && options.initialRoutes.length > 0 ? options.initialRoutes : ['/'];

  return render(
    <Provider store={storeMock}>
      <ThemeProvider theme={theme}>
        <MemoryRouter initialEntries={initialRoutes}>
          <Switch>
            <Route path={options.pathname} render={() => children} />
          </Switch>
        </MemoryRouter>
      </ThemeProvider>
    </Provider>,
  );
};
