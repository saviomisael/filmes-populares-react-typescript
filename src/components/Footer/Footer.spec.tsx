import { screen } from '@testing-library/react';
import { renderTheme } from 'utils/test-utils';
import { Footer } from '.';

describe('<Footer />', () => {
  it('should matches with snapshot', () => {
    renderTheme(<Footer />);

    const footer = screen.getByTestId('footer');

    expect(footer).toMatchSnapshot();
  });
});
