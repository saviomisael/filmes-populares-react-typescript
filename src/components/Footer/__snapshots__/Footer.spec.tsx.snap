// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should matches with snapshot 1`] = `
.c2 {
  color: #FFFFFF;
  font-size: 1.8rem;
}

.c1 {
  font-weight: 400;
  font-size: 1.8rem;
  color: #FFFFFF;
  text-align: left;
}

.c3 {
  width: 48.9rem;
}

.c0 {
  background-color: #F11C27;
  height: 25vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 2.2rem;
  column-gap: 2.2rem;
}

<footer
  class="c0"
  data-testid="footer"
>
  <p
    class="c1"
  >
    Dados providos por:
  </p>
  <a
    class="c2"
    href="https://developers.themoviedb.org/3/getting-started/introduction"
    rel="noopener"
    target="_blank"
  >
    <img
      alt="The Movie DB logo"
      class="c3"
      src="tmdb-logo.svg"
    />
  </a>
</footer>
`;
